recon/src/main/java/org/hps/recon/utils/TrackClusterMatcherMinDistance.java

```diff
@@ -0,0 +1,753 @@
+                if(skipTracks.contains(track))
+                    continue;
+                if(otherTrack == track)
+                    continue;
+                Cluster othersmallestdrCluster = trackClusterPairs.get(otherTrack);
+                if(othersmallestdrCluster == smallestdrCluster)
+                {
+                    duplicateCluster = true;
+                    trackswDuplicateClusters.put(track, smallestdrCluster);
+                    trackswDuplicateClusters.put(otherTrack, othersmallestdrCluster);
+                }
+            }
+
+            double smallestdr = 99999.0;
+            Track smallestdrTrack = null;
+            if(trackswDuplicateClusters == null)
+                return trackClusterResidualsMap;
+            for(Track duptrack : trackswDuplicateClusters.keySet()){
+                double dr = trackClusterResidualsMap.get(duptrack).get(trackswDuplicateClusters.get(duptrack));
+                if(dr < smallestdr){
+                    smallestdr = dr;
+                    smallestdrTrack = duptrack;
+                }
+            }
+            for(Track duptrack : trackswDuplicateClusters.keySet()){
+                skipTracks.add(duptrack);
+                if(duptrack != smallestdrTrack){
+                    trackClusterResidualsMap.get(duptrack).remove(trackswDuplicateClusters.get(duptrack));
+                }
+            }
+        }
+        return trackClusterResidualsMap;
+    }
+
+    private Map<Track,Cluster>  getMinTrackClusterResidualMap(Map<Track, Map<Cluster, Double>> trackClusterResidualsMap){
+
+        //inputs a mapping of tracks with residuals for each possible cluster
+        //from all clusters in the map, for each track, match the cluster with
+        //the smallest position residual to that track
+        //build output map of track -> closest cluster
+        Map<Track,Cluster> Map = new HashMap<Track, Cluster>();
+        for(Track track : trackClusterResidualsMap.keySet()){
+            double smallestdr = 99999.0;
+            Cluster smallestdrCluster = null;
+            Map<Cluster, Double> cluster_dr_Map = trackClusterResidualsMap.get(track);
+            for(Cluster c : cluster_dr_Map.keySet()){
+                double dr = cluster_dr_Map.get(c);
+                if(dr < smallestdr){
+                    // prefer using GBL tracks to correct (later) the clusters, for some consistency:
+                    if (track.getType() >= 32 || !this.clusterToTrack.containsKey(smallestdrCluster)) {
+                        this.clusterToTrack.put(smallestdrCluster, track);
+                    }
+
+                    smallestdr = dr;
+                    smallestdrCluster = c;
+                }
+            }
+            Map.put(track, smallestdrCluster);
+        }
+        return Map;
+    }
+
+    private boolean isTrackInEcal(double trackx, double tracky){
+
+        //Define first order Ecal geometry --> assumed square here,
```

Comment:
Is it possible to read this numbers from geometry?

Reply:
Not sure how to do this. This really isn't necessary to keep probably, I think I should remove. It was only really used as a sanity check